README.md

```diff
@@ -57,8 +70,38 @@ sql.Open("CovenantSQL", dbURI)
     - Each Database has its own independent distributed engine.
     - Mainly responsible for: database storage & encryption, query processing & signature, efficient indexing.
 
-## How it works
-[How CovenantSQL works(video)](https://youtu.be/2Mz5POxxaQM?t=106)
+### Consensus Algorithm
+
+CQL supports 2 kinds of consensus algorithm:
+
+1. DPoS (Delegated Proof-of-Stake) is applied in `Eventually consistency mode` database and also `Layer 1 (Global Consensus Layer)` in BlockProducer. CQL miners pack all SQL queries and its signatures by the client into blocks thus form a blockchain. We named the algorithm [`Xenomint`](https://github.com/CovenantSQL/CovenantSQL/tree/develop/xenomint). 
+2. BFT-Raft (Byzantine Fault-Toleranted Raft)<sup>[bft-raft](#bft-raft)</sup> is applied in `Strong consistency mode` database. We named our implementation [`Kayak`](https://github.com/CovenantSQL/CovenantSQL/tree/develop/kayak).  The CQL miner leader does a `Two-Phase Commit` with `Kayak` to support `Transaction`.<sup>[transaction](#transaction)</sup>
+
+CQL database consistency mode and node count can be selected in datebase creation with command  `cql create '{"UseEventualConsistency": true, "Node": 3}'`
+
+## Comparison
+
+|                              | Ethereum          | Hyperledger Fabric       | Amazon QLDB   | CovenantSQL                                                  |
+| ---------------------------- | ----------------- | ------------------------ | ------------- | ------------------------------------------------------------ |
+| **Dev language**             | Solidity          | Chaincode   (Go, NodeJS) | ?             | Python, Golang, Java, PHP, NodeJS, MatLab                    |
```

Comment:
[Ethereum flavored WebAssembly]( https://github.com/ewasm/design) could add to Ethereum smart contract dev language.